spec/integration/modules/write_file_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt_spec/conn'
+require 'bolt_spec/files'
+require 'bolt_spec/integration'
+require 'bolt_spec/run'
+
+describe 'running a plan with write_file' do
+  include BoltSpec::Conn
+  include BoltSpec::Files
+  include BoltSpec::Integration
+  include BoltSpec::Run
+
+  let(:filename) { 'hello.txt' }
+  let(:content) { 'Hello, world!' }
+  let(:params) { %W[target=#{transport} content=#{content} destination=#{filename}] }
+  let(:modulepath) { File.expand_path(File.join(__dir__, '../../fixtures/modules')) }
+  let(:inventory) { conn_inventory.merge(config) }
+  let(:config) do
+    {
+      'config' => {
+        'ssh' => { 'host-key-check' => false },
+        'winrm' => { 'ssl' => false }
+      }
+    }
+  end
+
+  shared_examples 'when writing a file from a plan' do
+    # Delete the uploaded file after each test
+    after(:each) do
+      result = run_command("#{remove_cmd} #{filename}", transport.to_s, inventory: inventory)
+      expect(result.first['status']).to eq('success')
+    end
+
+    it 'writes a file to the destination' do
```

Comment:
I'm not sure if we needed any other tests for this function besides this one, since pretty much all of the logic is contained in `file::write` and `upload_file` and they have their own tests.